src/codec/decoder/audio.rs

```diff
@@ -130,3 +138,38 @@ impl AsMut<Context> for Audio {
 		&mut self.0
 	}
 }
+
+pub struct AudioFrameIter<'a, 'b> {
+	audio: &'a mut Audio,
+	packet: &'b mut packet::Packet,
+}
+
+impl<'a, 'b> Iterator for AudioFrameIter<'a, 'b> {
+    type Item = Result<Option<frame::Audio>, Error>;
+
+    fn next(&mut self) -> Option<Result<Option<frame::Audio>, Error>> {
+		unsafe {
+			if !self.packet.is_empty() {
+				let mut out = frame::Audio::empty();
```

Comment:
To avoid this, you could create and put the frame inside the `AudioFrameIter` itself, then you wouldn't need to create a new one on each iteration.